tests/andromede/test_ac_link.py

```diff
@@ -0,0 +1,385 @@
+            definition=-var("flow"),
+        ),
+        PortFieldDefinition(
+            port_field=PortFieldId("port2", "flow"),
+            definition=var("flow"),
+        ),
+    ],
+    binding_constraints=[
+        Constraint(
+            name="AC flow",
+            expression=var("flow")
+            == 1
+            / param("reactance")
+            * (
+                port_field("port1", "angle")
+                + var("shift")
+                - port_field("port2", "angle")
+            ),
+        )
+    ],
+)
+
+
+def test_ac_network_no_links():
+    database = DataBase()
+    database.add_data("D", "demand", ConstantData(100))
+
+    database.add_data("G", "p_max", ConstantData(100))
+    database.add_data("G", "cost", ConstantData(30))
+
+    node = Node(model=AC_NODE_MODEL, id="N")
+    demand = create_component(
+        model=DEMAND_MODEL,
+        id="D",
+    )
+
+    gen = create_component(
+        model=GENERATOR_MODEL,
+        id="G",
+    )
+
+    network = Network("test")
+    network.add_node(node)
+    network.add_component(demand)
+    network.add_component(gen)
+    network.connect(PortRef(demand, "balance_port"), PortRef(node, "injections"))
+    network.connect(PortRef(gen, "balance_port"), PortRef(node, "injections"))
+
+    scenarios = 1
+    problem = build_problem(network, database, TimeBlock(1, [0]), scenarios)
+    status = problem.solver.Solve()
+
+    assert status == problem.solver.OPTIMAL
+    assert problem.solver.Objective().Value() == 3000
+
+
+def test_ac_network():
+    database = DataBase()
+    database.add_data("D", "demand", ConstantData(100))
+
+    database.add_data("G", "p_max", ConstantData(100))
+    database.add_data("G", "cost", ConstantData(35))
+
+    database.add_data("L", "reactance", ConstantData(1))
+
+    node1 = Node(model=AC_NODE_MODEL, id="1")
+    node2 = Node(model=AC_NODE_MODEL, id="2")
+    demand = create_component(
+        model=DEMAND_MODEL,
+        id="D",
+    )
+
+    gen = create_component(
+        model=GENERATOR_MODEL,
+        id="G",
+    )
+
+    link = create_component(
+        model=AC_LINK_MODEL,
+        id="L",
+    )
+
+    network = Network("test")
+    network.add_node(node1)
+    network.add_node(node2)
+    network.add_component(demand)
+    network.add_component(gen)
+    network.add_component(link)
+    network.connect(PortRef(demand, "balance_port"), PortRef(node1, "injections"))
+    network.connect(PortRef(gen, "balance_port"), PortRef(node2, "injections"))
+    network.connect(PortRef(link, "port1"), PortRef(node1, "links"))
+    network.connect(PortRef(link, "port2"), PortRef(node2, "links"))
+
+    scenarios = 1
+    problem = build_problem(network, database, TimeBlock(1, [0]), scenarios)
+    status = problem.solver.Solve()
+
+    assert status == problem.solver.OPTIMAL
+    assert problem.solver.Objective().Value() == 3500
+
+    for variable in problem.solver.variables():
+        if "balance_port_from" in variable.name():
```

Comment:
I don't understand where balance_port_from and balance_port_to variables are defined. Shouldn't it be variable "flow" ?

Reply:
You're right, the assertions here are wrong

Reply:
Fixed it in the last commits: it's now the flow variable which is checked, following your advice.